WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -551,6 +551,27 @@ final class EditableOrderViewModel: ObservableObject {
         resetAddressForm()
     }
 
+    func addTaxRateAddressToOrder(taxRate: TaxRate) {
+        guard let taxBasedOnSetting = taxBasedOnSetting else {
+            return
+        }
+
+        let address = Address.from(taxRate: taxRate)
+        let input: OrderSyncAddressesInput
+        switch taxBasedOnSetting {
+        case .customerBillingAddress:
+            input = OrderSyncAddressesInput(billing: address, shipping: nil)
+        case .customerShippingAddress:
+            input = OrderSyncAddressesInput(billing: nil, shipping: address)
+        default:
+            // Do not add address if the taxes are not based on the customer's addresses
+            return
+        }
+
+        orderSynchronizer.setAddresses.send(input)
+        resetAddressForm()
```

Comment:
I'm not sure to fully understand what `resetAddressForm` does here, I think I'm missing some context: 

This seems to create a `CreateOrderAddressFormViewModel` with the existing order data, then if there's an address update it sends the data to the ordersynchronizer, but we're already doing this with `orderSynchronizer.setAddresses.send(input)` and since `onAddressUpdate` seems to be only invoked on the view models' init, it shouldn't be called because we're holding the same instance of the EditableOrder.

Reply:
`resetAddressForm` creates a new `CreateOrderAddressFormViewModel` with the new data, so it can be shown here in the editable order details and once the user navigates to the Address Form. It also sets the closure for the future case when the user changes that address in the Address Form, note that that will only be called in the future.

> since onAddressUpdate seems to be only invoked on the view models' init

`onAddressUpdate` is called whenever the address is changed in the form, e.g. [here](https://github.com/woocommerce/woocommerce-ios/blob/0e92b8d52b0e54032d0f8e86efd89789b9acc766/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/CustomerSection/CreateOrderAddressFormViewModel.swift#L88)